### PR TITLE
ZER-79 - Added Create Calculator page on admin side

### PR DESCRIPTION
**Pull request by halyna-m**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by andriibureviy — APPROVED** [review]

**Review by Nazarii-Makohin — APPROVED** [review]

File: app/controllers/admins/calculators_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                    notice: 'Calculator was successfully created.'
+      else
+        render action: 'new'
+      end
+    end
+
+    def update
+      if @calculator.update(calculator_params)
+        redirect_to edit_admins_calculator_path(@calculator.id),
+                    notice: 'Successfully Updated'
```
Comment (maksymst) [inline app/controllers/admins/calculators_controller.rb:25]:
```suggestion
                    notice: 'Calculator has been successfully updated'
```

File: app/controllers/admins/calculators_controller.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    def new
+      @calculator = Calculator.new
+    end
+
+    def create
+      @calculator = Calculator.new(calculator_params)
+      if @calculator.save
+        redirect_to edit_admins_calculator_path(@calculator.id),
+                    notice: 'Calculator was successfully created.'
```
Comment (maksymst) [inline app/controllers/admins/calculators_controller.rb:16]:
```suggestion
                    notice: 'Calculator has been successfully created.'
```

File: app/controllers/admins/calculators_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        render action: 'new'
+      end
+    end
+
+    def update
+      if @calculator.update(calculator_params)
+        redirect_to edit_admins_calculator_path(@calculator.id),
+                    notice: 'Successfully Updated'
+      else
+        flash[:error] = 'Invalid'
```
Comment (maksymst) [inline app/controllers/admins/calculators_controller.rb:27]: you display errors by using `@calculator.errors`. why do you need this?

File: app/controllers/admins/calculators_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Admins
+  class CalculatorsController < ApplicationController
+    before_action :find_calculator, except: %i[new create calculator_params]
```
Comment (maksymst) [inline app/controllers/admins/calculators_controller.rb:5]: `except` and `include` options contain only `controller actions`

**Review by maksymst — CHANGES_REQUESTED** [review]
You've added a new validation in the Calculator model. It requires a unit test. Also, please add capybara test

**Review by andriibureviy — APPROVED** [review]

**Review by Nazarii-Makohin — DISMISSED** [review]

File: spec/features/admin/create_calculator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      click_button CREATE_CALCULATOR_BUTTON
+      expect(page).to have_content('Calculator has been successfully created')
+    end
+  end
+
+  context 'when user clicks button Create calculator' do
+    it 'redirects to Edit calculator page' do
+      fill_in 'Name', with: 'Calculator2'
+      click_button CREATE_CALCULATOR_BUTTON
+      expect(page).to have_current_path("/admins/calculators/1/edit")
```
Comment (maksymst) [inline spec/features/admin/create_calculator_spec.rb:24]: calculator identifier in the url can not be static

**Review by maksymst — CHANGES_REQUESTED** [review]

**Review by andriibureviy — APPROVED** [review]

**Review by Nazarii-Makohin — APPROVED** [review]

**Review by maksymst — APPROVED** [review]

**Pull request merged**